client/src/pages/JoinPage/index.tsx

```diff
@@ -22,19 +26,74 @@ const JoinPage = ({
   const { stream, setStream } = useContext(DeviceContext);
   const { room_id } = useParams<{ room_id: string }>();
   const [showNotification, setShowNotification] = useState(false);
+  const [notificationText, setNotificationText] = useState("");
+
+  // The password text.
+  const [password, setPassword] = useState("");
+  // A boolean to decide whether to disable the password input or not.
+  const [requirePassword, setRequirePassword] = useState(false);
```

Comment:
I personally think `is` prefix would be a better naming for boolean variables.
for this, `isPasswordRequired`, `setPasswordRequired` in this case

Reply:
Ah I see yeah that makes sense. I agree